src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -64,30 +64,423 @@ namespace {
+        if ((modTokens[0].type() == Opm::UDQTokenType::binary_op_sub)) {
+            if (startParen.size() > 0) {
+                // if followed by start_paren linked to end_paren before end of data
+                // set flag and remove from operator list because it is considered as a highest precedence operator
+                // unless () go from token 2 two the end of expression
+                if ((startParen[0] == 1)  && (endParen[0] < modTokens.size()-1)) {
+                    leadChgSgn = true;
+                }
+            } else {
+                // set flag and remove from operator list
+                leadChgSgn = true;
+            }
+            if (leadChgSgn) {
+                // remove from operator list because it is considered as a highest precedence operator and is
+                // therefore not a normal "binary_op_sub" operator
+                std::vector<Opm::UDQToken> temp_high_lev_op(highLevOp.begin()+1, highLevOp.end());
+                highLevOp = temp_high_lev_op;
+            }
+        } else if (startParen.size() >= 1) {
+        //
+        // check for leading start_paren combined with end_paren at end of data
+            if ((startParen[0] == 0) && (endParen[0] == modTokens.size()-1)) {
+                //
+                // remove leading and trailing ()
+                const std::vector<Opm::UDQToken> modTokens_red(modTokens.begin()+1, modTokens.end()-1);
+                noLeadOpenPar +=1;
+                //
+                // recursive call to itself to re-interpret the token-input
+
+                substOuterParentheses substOpPar = substitute_outer_parenthesis(modTokens_red, noLeadOpenPar, leadChgSgn);
+                highLevOp = substOpPar.highestLevOperators;
+                substTok = substOpPar.substitutedTokens;
+                noLeadOpenPar = substOpPar.noleadingOpenPar;
+                leadChgSgn = substOpPar.leadChangeSign;
+            }
+        }
+        // interpretation of token input is completed return resulting object
+
+        return {
+            highLevOp,
+            substTok,
+            noLeadOpenPar,
+            leadChgSgn
+        };
+    }
+
+    // Categorize function in terms of which token-types are used in formula
+    //
+    // The define_type is (-) the location among a set of tokens of the "top" of the parse tree (AST - abstract syntax tree)
+    // i.e. the location of the lowest precedence operator relative to the total set of operators, functions and open-/close - parenthesis
+    int define_type(const std::vector<Opm::UDQToken>& tokens)
+    {
+        int def_type = 0;
+        int noLeadOpenPar = 0;
+        bool leadChgSgn = false;
+
+        //
+        // analyse the expression
+
+        substOuterParentheses expr = substitute_outer_parenthesis(tokens, noLeadOpenPar, leadChgSgn);
+
+        //
+        // loop over high level operators to find operator with lowest precedence and highest index
+
+        int curPrec  = 100;
```

Comment:
The numerical values of `100` looks suspiciously as "large enough"?

Reply:
Yes - this is a number larger than the highest operator precedence.